### ISSUE TITLE
regarding administration and hosting support

### ISSUE THREAD

**Issue by gislars** [umap-project/umap#578]
Hello uMap Team,

I had the bold idea to host another independent setup of uMap in Germany. Now I was asked why I would put effort in additional work instead of helping each other.

I know there is no support and no paid maintainers, that is absolutely fine. I just wanted to ask if there is something other community members could help with. E.g. if a server upgrade would make the online service more reliable. Or a time consuming task which could better solved with combined effort.  

I would also appreciate information about the general status or road map (if there is any) for keeping the online service alive.

Thanks in advance!

Lars

**Comment by Nakaner** [comment]
Lars asked FOSSGIS e.V., the German OSMF local chapter, for a grant to pay a server running a independent instance of uMap this week. I suggested to offer support (both work and money) for a server upgrade of the French uMap instance and/or to try some kind of two-server setup with one server being the master (writing) and one being the slave (reading).

What setup does the French uMap use? Are you interested in support?

**Comment by yohanboniface** [comment]
Hi @gislars and @Nakaner

Great proposals, thanks!

Let me try to answer you :)

## uMap status

I'd say the status is "it works without doing nothing so let's continue doing nothing" :s
Basically I'm the only maintainer of the code base (while @cquest and @jocelynj also help on maintaining the French servers), and I'm quite inactive on the project since more than one year (trying to be less on my computer and more outside, in short).
There are very few contributions, so the code base is quite sleeping.
I've in mind to do a sort of sprint to at least upgrade all the deps of uMap to the latest, and maybe merge the three modules (leaflet-storage, django-leaflet-storage and uMap itself) in just one (the idea is to make contributing easier).
There are [great](https://github.com/umap-project/umap/issues/468) [ideas](https://github.com/umap-project/umap/issues/465) on the [air](https://github.com/umap-project/Leaflet.Storage/pull/118), but I'm missing time (or desire to work at night like before) to help them emerge.
Here are the most common requests: https://github.com/umap-project/umap/issues?q=is%3Aopen+is%3Aissue+label%3Apopular

## About French setup

First thing: we are close to migrate to a new server.
It's now running on an old VM with python2.7, we want to move it to a brand new (dedicated) VM on a server we have, and also upgrade python to 3.6.
The new VM is already created, so now I need to do the migration itself.
Here are more infos about the currently running setup: https://wiki.openstreetmap.org/wiki/FR:Serveurs_OpenStreetMap_France

## About hosting uMap

The suggestion to join efforts on the French instance is a great one, and I clearly understand the starting point.
Of course, having a uMap instance with master/slave servers will make it more robust.
Though I'm not sure this is the best way to move forward.
Basically, giving more energy on the French uMap will make it more and more visible and thus more and more critical and costly to maintain.
I sort of think it's more sustainable to have various instances, each of them being maintained by a different team, with different funding, availability, public, SLA policy, etc.
But I'll let @cquest and @jocelynj say their vision given they also help maintain the French server.
So my preference would be to have new instance, but the discussion is open, and I'll be available to whatever the scenario chosen, to help by coding or better documenting or just answering questions.

Glad to see you interest!

Yohan

**Comment by Marc-marc-marc** [comment]
my vision (as a helper of osm-fr infra) : a pool of sysadmin is better than several "not-releated" sysadmin working on several instance mainly for the following reasons:
availability: when one sysadmin is unavailable, another can deal with the problem.
knowledge sharing: when a problem occurs on one server, it is likely that this kind of problem can also occur on other instances.
load balancing: having several servers responding to the same url and located in separate datacenters is the best way to facilitate load balancing. updating a server in this kind of config is easy:  remove it from the dns, working "offline", testing and adding it to the dns. the migration of an instance is on the other hand more complicated if one wants to avoid cutting the service.
I can free up some time to help migrate the current osm-fr umap server to the new one

**Comment by yohanboniface** [comment]
> I've in mind to do a sort of sprint to at least upgrade all the deps of uMap to the latest, and maybe merge the three modules (leaflet-storage, django-leaflet-storage and uMap itself) in just one (the idea is to make contributing easier).

Work in progress https://github.com/umap-project/umap/pull/583
